pkg/conditions/conditions.go

```diff
@@ -0,0 +1,55 @@
+package conditions
+
+import (
+	"fmt"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	ProcessingCombinationsCondition = metav1.Condition{
+		Type:    "InProgress",
+		Status:  "Unknown",
+		Message: "Combination is not yet ready.",
+		Reason:  "Processing",
+	}
+	TemplateNotFoundCondition = metav1.Condition{
+		Type:    "Invalid",
+		Status:  "False",
+		Message: "Combination cannot be processed as template was not retrievable.",
+		Reason:  "TemplateNotFound",
+	}
+	TemplateBodyInvalid = metav1.Condition{
+		Type:    "Invalid",
+		Status:  "False",
+		Message: "Combination cannot be processed as template body was not valid.",
+		Reason:  "TemplateBodyInvalid",
+	}
+	ManifestGenerationFailed = metav1.Condition{
+		Type:    "Finished",
+		Status:  "False",
+		Message: "Generation of the manifest combinations failed.",
+		Reason:  "EvaluationInvalid",
+	}
+	ProccessedCombinationsCondition = metav1.Condition{
```

Comment:
Hmm I think we may want to investigate reworking (or treating status condition management as a follow-up entirely) - my initial thought is that we'd want to avoid harcoding certain metav1.Condition variables for each condition matrix array element. We could instead just use several constant variables:

```go
const (
    EvaluationInvalidReason = "EvaluationInvalid"
    TemplateBodyInvalidReason = "TemplateBodyInvalid"
    ...
```

Typically I've seen this types of variables be thrown into their own dedicated `types` packages in projects in the k8s ecosystem.

Reply:
We could then install factory methods for building up certain conditions, or setter-esq helper functions to enable the configuration we need when reconciling combo APIs.

Reply:
Just coming back to this now: I'd like to move any of those Reasons/Phase/etc. as constant types in the API package(s) (e.g. a types.go) vs. constant variables for full blown metav1.Condition.

Reply:
Overhauled Status management with the most recent commit. PTAL!

https://github.com/operator-framework/combo/pull/61/commits/646a11e6e1e7b0910bbffbb81976b7dbfdd3bcd0